server/game/gameSystems/ReadyResourcesSystem.ts

```diff
@@ -0,0 +1,41 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardType, Location, WildcardCardType } from '../core/Constants';
+import Player from '../core/Player.js';
+import * as EnumHelpers from '../core/utils/EnumHelpers';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem.js';
+
+
+export interface IReadyResourcesSystemProperties extends IPlayerTargetSystemProperties {
+    amount?: number
+}
+
+export class ReadyResourcesSystem<TContext extends AbilityContext = AbilityContext, TProperties extends IReadyResourcesSystemProperties = IReadyResourcesSystemProperties> extends PlayerTargetSystem<TContext, TProperties> {
```

Comment:
Since readying and exhausting resources are so similar, we probably want to have a base class that implements the core logic for both and then extend it with subclasses for each. Take a look at the ReadySystem and ExhaustSystem for an example of how to do this with the parent class ExhaustOrReadySystem.

Reply:
There's really not much shared between them(as it stands, looking at PayResourceCostSystem.ts)(just having an "amount" property). It doesn't seem worth the extra file to me.

Reply:
Oh yeah that's fair actually, I guess since this system is targeting players there are far fewer checks to do so the system is much simpler. Seems good to me.